plugins/unicodelib/unicodelib.cpp

```diff
@@ -719,6 +724,48 @@ unsigned unicodeEditDistanceV4(UnicodeString & left, UnicodeString & right, unsi
     return da[mask(leftLen-1)][rightLen-1];
 }
 
+void normalizationFormCheck(UnicodeString & source, const char * form)
+{
+    UErrorCode errorCode = U_ZERO_ERROR;
+    if (form[2] == nfc[2])
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[2] == nfd[2])
+    {
+        const Normalizer2 * no = Normalizer2::getInstance(NULL, "nfc", UNormalization2Mode::UNORM2_DECOMPOSE, errorCode);
+        source = no->normalize(source, errorCode);
+    }
+    else if (form[3] == nfkc[3])
```

Comment:
This could access invalid memory - e.g., if "XYZ" is passed as the form.  It would be clearer to explicitly compare against the characters.